### ISSUE TITLE
"blocked-plugins" config setting

### ISSUE THREAD

**Issue by HexedHero** [SlimeDog/NetworkInterceptor#19]
**Is your feature request related to a problem? Please describe.**
N/A

**Describe the solution you'd like**
Add a blocked-plugin config setting to block all connections from a set of plugins. The problem is that you cannot block a plugin without blocking others using the allow list.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by SlimeDog** [comment]
There are several possible interpretations of this request. I'm thinking to introduce an `untrusted-plugins` node, something like
```
# Untrusted plugins
# No network connections by the listed plugin(s) will be allowed, period.
# More than one plugin may be listed.
#
# Remove [] on the next line, and add the following line (without the # comment).
# - pluginName
untrusted-plugins: []
```
but I want to be sure that we address your use case. Please give a specific example with real targets if possible (fake plugin names if you prefer).

**Comment by SlimeDog** [comment]
BTW, we accept PRs, if you want to give it a go yourself.

**Comment by HexedHero** [comment]
> There are several possible interpretations of this request. I'm thinking to introduce an `untrusted-plugins` node, something like
> 
> ```
> # Untrusted plugins
> # No network connections by the listed plugin(s) will be allowed, period.
> # More than one plugin may be listed.
> #
> # Remove [] on the next line, and add the following line (without the # comment).
> # - pluginName
> untrusted-plugins: []
> ```
> 
> but I want to be sure that we address your use case. Please give a specific example with real targets if possible (fake plugin names if you prefer).

That is pretty much what I was after.

My issue was this:
PluginA and PluginB connects to example.com
I want to stop PluginA from connecting but if I blacklist that example.com it'll also stop PluginB too which I want to connect.
I guess I could add PluginB to the trusted list and block example.com but I'm not sure if that works and isn't user-friendly/confusing.

**Comment by SlimeDog** [comment]
> I guess I could add PluginB to the trusted list and block example.com but I'm not sure if that works and isn't user-friendly/confusing.

That does work, which is why I didn't introduce `untrusted-plugins` in the initial rewrite. I agree that an `untrusted-plugins` node is less confusing, and therefore desirable.

Were you going to make a PR, or should we?

**Comment by SlimeDog** [comment]
Let's go with `blocked-plugins` -- no negative judgment, just a statement of condition.

```
# Blocked plugins
# No network connections by the listed plugin(s) will be allowed, period.
# More than one plugin may be listed.
#
# Remove [] on the next line, and add the following line (without the # comment).
# - pluginName
blocked-plugins: []
```

**Comment by SlimeDog** [comment]
I created `branch=blocked-plugins` and updated the three configuration files.

**Comment by SlimeDog** [comment]
Implemented in 3.2.1. Published for Spigot/Paper, Bungee/Waterfall, and Universal.
